app/serializers/concerns/topic_tags_mixin.rb

```diff
@@ -23,6 +24,16 @@ def tags_descriptions
       .compact
   end
 
+  def tags_groups
+    all_tags
+      .each
+      .with_object({}) do |tag, acc|
+        acc[tag.name] = DiscourseTagging.tag_groups_for(tag) &
+          DiscourseTagging.visible_tag_groups_for(scope)
+      end
```

Comment:
These `DiscourseTagging` calls both involve calls to redis (`Discourse.cache`). So, when this serializer is used to render a topic list, it's potentially going to result in thousands of redis calls.